qutebrowser/browser/webengine/webenginesettings.py

```diff
@@ -392,6 +405,26 @@ def _init_profile(profile: QWebEngineProfile) -> None:
     _global_settings.init_settings()
 
 
+def _clear_webengine_permissions_json():
+    """Remove QtWebEngine's persistent permissions file, if present.
+
+    We have our own permissions feature and don't integrate with their one.
+    This only needs to be called when you are on Qt6.8 but PyQt<6.8, since if
+    we have access to the `setPersistentPermissionsPolicy()` we will use that
+    to disable the Qt feature.
+    This needs to be called before we call `setPersistentStoragePath()`
```

Comment:
Because we initialize the profile as `QWebEngineProfile("Default")` it's never an off the record one. It looks like the only thing that changes after calling `setPersistentStoragePath()` is that `persistentStoragePath()` returns the path you put in. I don't see a convenient check we can do here, but this case is covered by the new test in `test_invocations` and this helper method is called from immediately before we call `setPersistentStoragePath()`, so hopefully it's not too problematic.